File: PCD_CF849_Div4/D_DistinctSplit.py
t=int(input())
while t:
    n=int(input())
    s=input()
    count={}
    for i in range(len(s)):
        if s[i] in count:
            count[s[i]]+=1
        else:
            count[s[i]]=1
    ans=len(count)
    temp={}
    for i in range(len(s)):
        if s[i] in temp:
            temp[s[i]]+=1
        else:
            temp[s[i]]=1
        count[s[i]]-=1
        if(count[s[i]]==0):
            del count[s[i]]
        curr=len(count)+len(temp)
        ans=max(ans, curr)
    print(ans)
    t-=1

File: PCD/PCD_CF849_Div4/D_DistinctSplit.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int t;
    cin>>t;
    while(t--){
        int n;
        cin>>n;
        string s;
        cin>>s;
        map<char, int> count;
        for(int i=0; i<n; i++){
            count[s[i]]++;
        }
        map<char, int> temp;
        int ans=count.size()+temp.size();
        for(int i=0; i<n; i++){
            temp[s[i]]++;
            count[s[i]]--;
            if(count[s[i]]==0){
                count.erase(s[i]);
            }
            int curr=count.size()+temp.size();
            ans=max(ans, curr);
        }
        cout<<ans<<endl;
    }
    return 0;
}